src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -133,6 +133,14 @@ impl LogicalAgg {
                 TableCatalogBuilder::new(self.ctx().inner().with_options.internal_table_subset());
             let mut column_mapping = vec![];
 
+            let mut stored_upstream_idx = self
+                .group_key()
+                .iter()
+                .cloned()
+                .chain(sort_keys.iter().map(|&(_, idx)| idx))
+                .collect_vec();
+            stored_upstream_idx.sort();
+
```

Comment:
What about:

```rust
            let mut included_upstream_indices = HashSet::new();
            let mut add_column = |upstream_idx, order_type| {
                if !included_upstream_indices.contains(&upstream_idx) {
                    let tb_column_idx =
                        internal_table_catalog_builder.add_column(&in_fields[upstream_idx]);
                    column_mapping.push(upstream_idx);
                    if let Some(order_type) = order_type {
                        internal_table_catalog_builder.add_order_column(tb_column_idx, order_type);
                    }
                    included_upstream_indices.insert(upstream_idx);
                }
            };

            for &idx in self.group_key() {
                add_column(idx, Some(OrderType::Ascending));
            }
            for (order_type, idx) in sort_keys {
                add_column(idx, Some(order_type));
            }
            for &idx in &in_pks {
                add_column(idx, Some(OrderType::Ascending));
            }
            for idx in include_keys {
                add_column(idx, None);
            }
```

May need to rerun `do-apply-planner-test`.

Reply:
> What about:
> 
> ```rust
>             let mut included_upstream_indices = HashSet::new();
>             let mut add_column = |upstream_idx, order_type| {
>                 if !included_upstream_indices.contains(&upstream_idx) {
>                     let tb_column_idx =
>                         internal_table_catalog_builder.add_column(&in_fields[upstream_idx]);
>                     column_mapping.push(upstream_idx);
>                     if let Some(order_type) = order_type {
>                         internal_table_catalog_builder.add_order_column(tb_column_idx, order_type);
>                     }
>                     included_upstream_indices.insert(upstream_idx);
>                 }
>             };
> 
>             for &idx in self.group_key() {
>                 add_column(idx, Some(OrderType::Ascending));
>             }
>             for (order_type, idx) in sort_keys {
>                 add_column(idx, Some(order_type));
>             }
>             for &idx in &in_pks {
>                 add_column(idx, Some(OrderType::Ascending));
>             }
>             for idx in include_keys {
>                 add_column(idx, None);
>             }
> ```
> 
> , which will make the code cleaner.
> 
> May need to rerun `do-apply-planner-test`.

It's incorrect when we call `add_column(x, None)` before calling `add_column(x, Some(_))`

Reply:
> What do you mean by calling `add_column(x, None)` before calling `add_column(x, Some(_))`.🤔 The logic is not changed compared to the previous code.

I mean if someone misuses this `add_column`, and he calls `add_column(x, None)` before calling `add_column(x, Some(_))`

Reply:
But `add_column` is a temporary closure. It's not to hide the implementation, but just to simplify code. So devs should know what `add_column` does, just like before.